api/python/provisioner/commands/validator/validator.py

```
@@ -235,6 +242,8 @@ def validate(self, path: Path):
             raise ValidationError(reason=f"'{path}' is not a directory")
 
         if self.files_scheme:
+            if self.empty and not any(path.iterdir()):
```

If empty is false and no dirs makes sense to fail fast right here